### DISCUSSION TITLE
Supabase Magic Link with Expo React Native

### DISCUSSION THREAD

**Discussion by myhendry** [supabase/supabase#6698] [Questions] [answered]
possible to setup magic link with expo react native?

if so, how to redirect on mobile?

I know how to set up magic link on next js but now wondering if its possible to set up magic link using expo react native

Thanks

**Comment by cloudorbush** [comment]
[Here's](https://github.com/kelvinpompey/expo-supabase-user-management) an example that might help you.

**Reply by ammarfaris** [reply]
thanks for sharing @vbylen ; is this working for ios and android too? i tried it but it does not seem to work. thanks

**Reply by BoavistaLudwig** [reply]
any updates on this???

**Comment by Csierram96** [comment]
any updates?

**Comment by seanbenk** [comment]
Using the latest version of `@supabase/supabase-js` (`^2.26.0` at the time of writing), the unfortunately not as seamless user experience solution for me was to expose the OTP from the magic link email template using the `{{ .Token }}` field (https://supabase.com/docs/guides/auth/auth-email-templates). 
Once the user has requested an OTP, send them through to the classic "Enter your one time passcode" screen. You should then be able to call something along the lines of
```
async () => {
  const { data, error } = await supabase.auth.verifyOtp({
    email,
    token: OTP,
    type: "email",
  });
}
```
on OTP entry to sign them in (https://supabase.com/docs/reference/javascript/auth-verifyotp). 
Similarly I kept running into issues with the deeplink/extracting tokens from magic link => expo/react-native.

**Reply by kythonlk** [reply]
up

**Comment by thorwebdev** [comment, marked as answer]
You can see an example of using `expo-auth-session`'s `makeRedirectUri` here: https://github.com/bndkt/sharemystack/blob/main/apps/app/src/components/auth/SignInWithTwitter.tsx and the configuration here: https://github.com/bndkt/sharemystack/blob/main/apps/app/app.config.ts#L49

Note that you will also need to set up `associatedDomains` as outlined here: https://docs.expo.dev/guides/deep-linking/#native-apple-configuration 

I'm also working on creating some docs for this specifically. Will follow up here once ready.
